File: app/src/main/java/com/codemybrainsout/imageviewer/viewmodel/ExifViewModel.java
package com.codemybrainsout.imageviewer.viewmodel;

import android.databinding.BaseObservable;
import android.text.TextUtils;

import com.codemybrainsout.imageviewer.model.Exif;

/**
 *
 * Viewmodel required for binding exif data to layout_exif.xml
 *
 * Created by ahulr on 11-06-2017.
 */

public class ExifViewModel extends BaseObservable {

    private Exif exif;

    public ExifViewModel(Exif exif) {
        this.exif = exif;
    }

    public String getMake() {
        return TextUtils.isEmpty(exif.getMake()) ? "NA" : exif.getMake();
    }

    public String getModel() {
        return TextUtils.isEmpty(exif.getModel()) ? "NA" : exif.getModel();
    }

    public String getFocal() {
        return TextUtils.isEmpty(exif.getFocalLength()) ? "NA" : exif.getFocalLength() + "mm";
    }

    public String getAperture() {
        return TextUtils.isEmpty(exif.getAperture()) ? "NA" : "f/" + exif.getAperture();
    }

    public String getExposure() {
        return TextUtils.isEmpty(exif.getExposureTime()) ? "NA" : exif.getExposureTime() + "s";
    }

    public String getIso() {
        return exif.getIso() == null ? "NA" : "ISO " + String.valueOf(exif.getIso());
    }

    public String getLikes() {
        return exif.getLikes() == null ? "NA" : String.valueOf(exif.getLikes());
    }

    public String getDownloads() {
        return exif.getDownloads() == null ? "NA" : String.valueOf(exif.getDownloads());
    }
}

File: app/src/main/java/com/codemybrainsout/imageviewer/viewmodel/ExifViewModel.kt
package com.codemybrainsout.imageviewer.viewmodel

import android.text.TextUtils
import androidx.databinding.BaseObservable
import com.codemybrainsout.imageviewer.model.Exif

/**
 *
 * Viewmodel required for binding exif data to layout_exif.xml
 *
 * Created by ahulr on 11-06-2017.
 */
class ExifViewModel(exif: Exif) : BaseObservable() {
    private val exif: Exif
    val make: String?
        get() = if (TextUtils.isEmpty(exif.make)) "NA" else exif.make
    val model: String?
        get() = if (TextUtils.isEmpty(exif.model)) "NA" else exif.model
    val focal: String
        get() = if (TextUtils.isEmpty(exif.focalLength)) "NA" else exif.focalLength + "mm"
    val aperture: String
        get() = if (TextUtils.isEmpty(exif.aperture)) "NA" else "f/" + exif.aperture
    val exposure: String
        get() = if (TextUtils.isEmpty(exif.exposureTime)) "NA" else exif.exposureTime + "s"
    val iso: String
        get() = if (exif.iso == null) "NA" else "ISO " + exif.iso.toString()
    val likes: String
        get() = if (exif.likes == null) "NA" else exif.likes.toString()
    val downloads: String
        get() = if (exif.downloads == null) "NA" else exif.downloads.toString()

    init {
        this.exif = exif
    }
}